File: Solutions/10001st_prime.js
let primesArray = [2]
let current = 3

while(primesArray.length < 10001) {
  let bool = true
  for (let prime of primesArray) {
    if (current%prime === 0) {
      bool = false;
      break
    }
  }
  if(bool) {
    primesArray.push(current)
  }
  current ++
}

console.log(primesArray[10000])

File: Solutions/10001st_prime.cpp
#include<iostream>
using namespace std; 
int main()
{
    int  i=1,n,prime=2,count,flag,nprime; 
    cin>>n;
    while(i<=n)
    {
        flag=1;
        for(count=2;count<=prime-1;count++)
        {
            if(prime%count==0)  
            {
             flag=0;
             break;     
            }  
        }
        if(flag==1)
        {
        	nprime=prime;
            i++;
        }
        prime++;
    }
    cout<<nprime;
}